### PR TITLE
Theme update fixes

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): #

 - [WASM] update entire visual tree
 - apply style setters with correct precedence

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Tests/Windows_UI_Xaml/Given_ThemeResource.cs [hunk 1]
```diff
@@ -55,6 +55,22 @@ public async Task When_Theme_Changed_ApplicationPageBackground()
...
+		{
+			var page = new ThemeResource_Themed_Color_Page();
+			var app = UnitTestsApp.App.EnsureApplication();
+			app.HostView.Children.Add(page);
+
+			var button = page.TestButton;
+
+			Assert.AreEqual(Colors.Peru, (button.Foreground as SolidColorBrush).Color);
+
+			await SwapSystemTheme();
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/Windows_UI_Xaml/Given_ThemeResource.cs:69]: 🚀

File: src/Uno.UI/UI/Xaml/Application.cs [hunk 2]
```diff
@@ -199,7 +199,7 @@ private void SetRequestedTheme(ApplicationTheme requestedTheme)
 
 		private void OnRequestedThemeChanged()
 		{
-			if (Windows.UI.Xaml.Window.Current.Content is FrameworkElement root)
+			if (GetTreeRoot() is FrameworkElement root)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Application.cs:202]: I'm wondering if we should end up using the newer `XamlRoot` property

File: src/Uno.UI/UI/Xaml/DependencyObjectStore.cs [hunk 3]
```diff
@@ -1129,7 +1129,7 @@ internal void UpdateResourceBindings(bool isThemeChangedUpdate, ResourceDictiona
 					if (dict.TryGetValue(kvp.Value.ResourceKey, out var value, shouldCheckSystem: false))
 					{
 						wasSet = true;
-						SetValue(kvp.Key, BindingPropertyHelper.Convert(() => kvp.Key.Type, value));
+						SetValue(kvp.Key, BindingPropertyHelper.Convert(() => kvp.Key.Type, value), kvp.Value.Precedence);
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:1132]: I forgot to ask in the previous PR, are there ways to validate that in runtime/unit test ?

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/DependencyObjectStore.cs:1132]: depends, do you mean Uno.UI.RuntimeTests? It should be possible to run the tests there, now that `SwapSystemTheme()` can be implemented (or faked) by setting Window.Current.Content.RequestedTheme

File: src/Uno.UI/UI/Xaml/Application.cs (hunk 2 above)
Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Application.cs:202]: ...huh ...perhaps? Never noticed that one before. Not sure if it's applicable in this context.

**Pull request merged**